python/lsst/jointcal/ccdImage.cc

```diff
@@ -66,13 +65,11 @@ void declareCcdImageList(py::module &mod) {
 
     cls.def("sizeValidForFit", &CcdImageList::sizeValidForFit);
 
-    // Make this behave like a std::list, which it is, but we can't wrap children of it easily.
+    // Make this behave like a std::list (which it is but we can't wrap std::list subclasses trivially)
```

Comment:
I unraveled `CcdImageList` into `typedef std::list<std::shared_ptr<CcdImage> > CcdImageList;`, with no obvious ill effects. I'm going to punt on the question of how to manage the various `StarList`s as out of scope for this ticket and deal with it in DM-4043 (which I plan to schedule for next cycle).

I suspect we'll stick with `list` instead of `vector` here, since than the algorithm has a cleanup step that removes items from the middle of the list. But lets table that until DM-4043.